plugin/pkg/bastion/validator/admission.go

```diff
@@ -128,8 +128,22 @@ func (v *Bastion) Admit(ctx context.Context, a admission.Attributes, o admission
 		return apierrors.NewInvalid(gk, bastion.Name, field.ErrorList{field.Required(shootPath, "shoot is required")})
 	}
 
-	// ensure shoot exists
 	shootName := bastion.Spec.ShootRef.Name
+
+	if a.GetOperation() == admission.Update {
+		oldBastion, ok := a.GetOldObject().(*operations.Bastion)
+		if !ok {
+			return apierrors.NewBadRequest("could not convert old object to Bastion")
+		}
+
+		oldShootName := oldBastion.Spec.ShootRef.Name
+		if oldShootName != shootName {
+			fieldErr := field.Invalid(shootPath, shootName, "shootRef must not be changed")
+			return apierrors.NewInvalid(gk, bastion.Name, field.ErrorList{fieldErr})
+		}
+	}
```

Comment:
Why it was required to add such validation here?
I think validation for the immutability of the `shootRef.name` fields already exists.

https://github.com/gardener/gardener/blob/ca661c0502eb1f885a2ac3793913025788d5102d/pkg/apis/operations/validation/bastion.go#L85

Reply:
right, this validation is duplicative. I removed it in https://github.com/gardener/gardener/pull/5345/commits/d4e5712ddb313823cfbaa65a53cfcc8050acfe5d